lib/albacore/solution.rb

```diff
@@ -0,0 +1,37 @@
+require 'albacore/logging'
+require 'albacore/project'
+
+module Albacore
+  # a solution encapsulates the properties from a sln file.
+  class Solution
+    include Logging
+
+    attr_reader :path_base, :filename, :content
+
+    def initialize path
+      raise ArgumentError, 'solution path does not exist' unless File.exists? path.to_s
+      path = path.to_s unless path.is_a? String
+      @content = open(path)
+      @path_base, @filename = File.split path
+    end
+
+    def projects
+      project_paths.map { |path| File.join(@path_base, path) }
+                   .select { |path| File.file?(path) }
+                   .map {|path| Albacore::Project.new(File.absolute_path(path)) }
+    end
+
+    def project_paths
+      project_matches.map { |matches| matches[:location] }
```

Comment:
Let's make it better for multi-lingual sln files while we're at it and have any file ending with `proj`?